crates/sargon/src/signing/petition_types/general_role_with_hd_factor_instance.rs

```diff
@@ -12,10 +12,8 @@ impl TryFrom<(MatrixOfFactorInstances, RoleKind)>
 {
     type Error = CommonError;
 
-    fn try_from(value: (MatrixOfFactorInstances, RoleKind)) -> Result<Self> {
-        let matrix = value.0;
-
-        let (threshold_factors, threshold, override_factors) = match value.1 {
+    fn try_from((matrix, role): (MatrixOfFactorInstances, RoleKind)) -> Result<Self> {
```

Comment:
`(matrix, role): (MatrixOfFactorInstances, RoleKind)` this is cool, I did not know this was possible, thx for teaching me!